### PR TITLE
Quickstart review

### LINKED ISSUE #53: Quickstart back office documentation is outdated
cf http://kuzzle.io/guide/#setup-a-kuzzle-backoffice

### PR DESCRIPTION
**Pull request by xbill82**
Take a look at Luca's PC http://192.168.1.157:4567/#quickstart
Fixes #53, #63, #75 (and others)

* Proposed a new visual style that improves readability,
* re-written the Quickstart section.
* applied font face from the official style guidelines https://projects.invisionapp.com/boards/DP339D0QGXZA4#/5190948/158206618

### CODE REVIEW

**Comment by stafyniaksacha** [comment]
Yep nice work !

Two things: 
- the blocks in introduction part should be reworked to fit to the new screen section
- the background margins should be uninformed too

**Comment by dbengsch** [comment]
Nice work. I agree with @stafyniaksacha about intro blocks (and they could be responsive too)

**Comment by xbill82** [comment]
Check the updates out, @stafyniaksacha @dbengsch !

**Comment by melmag** [comment]
Color link have to be like this one: 
image
#3498db

Not like this:
image

We can apply the same color rules as back-office?
image
This part in "Skin" color ? (#E08E79)

And why titles are in Arial font ? We have to use Ubuntu (like said by our famous designer...) ^^

**Comment by xbill82** [comment]
@melmag Done - u can check at http://192.168.1.157:4567/#quickstart

**Comment by benoitvidis** [comment]
In one of my previous lives, I happened to work for a printing house, where I was told mixing text transformations at a same time was a bad practice. I cannot find strong material on the subject and I don't actually know if this is a real convention, but having the links at the same time 
* colorized
* bolded
* undelined

is definitely too much for me.

Just changing like most website (including Github) do, would be good enough to me, possibly having some underlying on the hover:
image

I suppose we can get some inputs from the office next door on this.

**Comment by benoitvidis** [comment]
I really appreciate the effort but I must admit some things do not work for me.

* The width limit is a bit disturbing, probably because of the left column and the full-width menu
* How this limited width would work with our 2 columns layout documentations (sdk & api)?
* Too much beige
* The separation between paragraphs is not clear 
* The notice blocks look to emphasized compared to the main text

That being said, I feel totally incompetent in any UX-related area. 

I am wondering if trying to get a consensus from developers based on this pr is the most efficient approach.
Wouldn't it be better to get some mock-ups from Gradientz that we could use for reference?

**Comment by melmag** [comment]
It's ok for me.
For the UX aspect, I think a meeting for regroup your needs is very important now.
We can't work in a hurry, especially with design and UX.
In my opinion, this version is the minimal acceptable for the quickstart.

**Comment by xbill82** [comment]
**TL;DR** @benoitvidis you're not incompetent. We should focus on copy instead of desing. Crappy docs lead to oblivion.

@benoitvidis feedback from "totally incompetent UX people" are the most valuable, most of the times. One cannot call a user pointing out problems an "incompetent" feedback. This is actually the feedback we need, so your remarks are totally welcome. We're all here to ship the best documentation possible and I agree with you we're not yet at the "decent" level.

Take a look at the design updates (link and header font styles) I made this morning, they should match your remarks.

Concerning the line-width:
1. We must agree that a documentation is meant to be _read_ by human eyes and that **long lines degrade readability**. So, to me, short _text_ lines are mandatory. Many documentations have this exact layout (e.g. [Vuejs Guide](https://vuejs.org/v2/guide/)) and it seems to work pretty well. Nonetheless, I think your remark should not be underestimated. The current design _does feel disturbing_, but I think this is due to another reason (still not clear to me, but deserves further investigation or even designer audit).
2. Point 1 doesn't imply we must apply the same `width` CSS rule to the `.content` class in the 2 columns layout doc. And this is for a simple reason: in 2 columns layout docs, text lines are already short. So, to me, **this line width must be applied to the Guide only**.
3. I do agree with you on "Too much beige" and "no clear paragraph separation". Don't know about the emphasis on notice blocks (deserves a try).

Having **Gradientz** ship a decent desing _all at once_ would be awesome. Yet, they seem to be busy on other subjects, but I do agree that the best thing to do is to **ask for a time slot**. @melmag could you provide us with a time estimation on "when" and "how much time"?
But something tells me it's going to take some time. In the meanwhile, I think that we can improve the docs with
* Some _light_ design improvements (i.e. being consistent with the guidelines and have decent readability, without going farther than that).
* Some _heavy_ **copywriting** improvements, since docs are made by developers for developers and I think it is our responsibility to document what we ship. I can work on this point but **I need real reviews on the text** and I have the feeling that we are focusing too much on the design here.

I truly believe that the docs are a determinant factor of product adoption. Surely not the only one, but crappy docs definitely lead to oblivion.

File: source/includes/quickstart/_index.md [hunk 1]
```diff
@@ -1,142 +1,216 @@
...
+<b>Docker and Docker-compose are not mandatory</b>. The Kuzzle Server can run outside a Docker container. This Quickstart tutorial uses Docker as it simplifies a lot the startup process. If you wish to run a Kuzzle Server without Docker, jump to the <a href="#manual-install">manual installation guide</a>.
 </aside>
 
-- Copy our [docker-compose.yml](https://raw.githubusercontent.com/kuzzleio/kuzzle/master/docker-compose.yml) file in a directory
+Thanks to Docker-compose, running a Kuzzle server is easy. Just grab the standard [docker-compose.yml](https://raw.githubusercontent.com/kuzzleio/kuzzle/master/docker-compose.yml) file and copy it into a directory of your choice.
 
 <aside class="notice">
-You can also retrieve this file directly with curl/wget : <br />
-<code>wget https://raw.githubusercontent.com/kuzzleio/kuzzle/master/docker-compose.yml</code>
+Getting a remote file is very easy with curl/wget : <br />
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: This is a personal opinion, but there are too many "easy" words to my liking :-)
And I don't think putting this part in a "notice" block is necessary.

Wouldn't it be simpler to write something like this?

    [...]file, copy it into a directory, and start Kuzzle:

    ```bash
    $ wget https://raw.githubusercontent.com/kuzzleio/kuzzle/master/docker-compose.yml
    $ docker-compose up
    ```

File: source/includes/quickstart/_index.md [hunk 2]
```diff
@@ -1,142 +1,216 @@
...
 </aside>
 
-- run:
+Then, run
 
 ```bash
 $ docker-compose up
 ```
 
+In your terminal window, you will see the Docker-compose output coming from the components of the Kuzzle architecture. After a few seconds, the output messages will stop flowing, showing the following one at the end:
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: What do you think of this formulation?

```
Your terminal will show the log messages of Kuzzle's components starting. After only a few seconds, you should see the following ready message appear: 
```

File: source/includes/quickstart/_index.md [hunk 3]
```diff
@@ -1,142 +1,216 @@
...
+
+```
+kuzzle_1         | [✔] Kuzzle server ready
+```
+
+To check that the Kuzzle Server is actually running, you can hit the main HTTP API endpoint by browsing the page <a href="http://localhost:7511/api/1.0">http://localhost:7511/api/1.0</a> or via cURL on the command line:
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: "To check that the Kuzzle Server is actually running" => this implies that it may not be running, even if a "ready" message has been printed. Not very reassuring :-)

What do you think of the following instead?

```
Your Kuzzle instance is now ready to be used. For instance, you can hit [...]
```

Comment (scottinet) [inline source/includes/quickstart/_index.md]: RC8 URLs have been simplified:

`<a href="http://localhost:7511/api/1.0">http://localhost:7511/api/1.0</a>`

=> 

`<a href="http://localhost:7511/">http://localhost:7511/</a>`

File: source/includes/quickstart/_index.md [hunk 4]
```diff
@@ -1,142 +1,216 @@
...
+
+```bash
+$ curl "http://localhost:7511/"
+```
+
+Kuzzle will respond you with a friendly hello message :)
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: There is no friendly hello message anymore. Instead you'll be hit with a ginormous wall of text, since this route now calls `server:info`

I suggest replacing your example with a "simple" realtime message publication (with the appropriate changes in this documentation to reflect that change):

```bash
$ curl -H "Content-Type: application/json" --data '{"hello": "world"}' "http://localhost:7511/hello/world?pretty=true"
{
  "status": 200,
  "error": null,
  "requestId": "df1e6638-2efb-4c56-9838-178a4cd5c6bd",
  "controller": "realtime",
  "action": "publish",
  "collection": "world",
  "index": "hello",
  "metadata": null,
  "headers": {},
  "result": {
    "published": true
  }
}
```

Or maybe with a simple `adminExists` request:

```bash
$ curl"http://localhost:7511/_adminExists?pretty=true"
{
  "status": 200,
  "error": null,
  "requestId": "6d1dc0d6-3274-4511-955c-e01875f2a5e6",
  "controller": "server",
  "action": "adminExists",
  "collection": null,
  "index": null,
  "metadata": null,
  "headers": {},
  "result": {
    "exists": false
  }
}
```

File: source/includes/quickstart/_index.md [hunk 5]
```diff
@@ -1,142 +1,216 @@
...
+
+You can use the <a href="http://kuzzle-backoffice.netlify.com/">publicly hosted Kuzzle Backoffice</a> or host your own (see next section). In both cases, you'll be able to <a href="#select-the-kuzzle-instance-to-connect-to">select the Kuzzle Server</a> you want to manage once the Backoffice starts up.
+
+### Host your own instance of the Kuzzle Backoffice
 
-Kuzzle has a backoffice which permit you to see, create, watch and manage everything you can do with kuzzle.
+Before proceeding, ensure that your system matches the following requisites:
 
-The backoffice is not mandatory to run kuzzle, and it's not incuded by default, the easiest way to use the kuzzle's backoffice is to add this section into your `docker-compose.yml` file:
+* A fairly recent version of **NodeJS** (i.e. v4.3.1+) should be installed on your system (<a href="https://nodejs.org/en/download/">instructions here</a>),
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: (i.e. v4.3.1+) => (e.g.  v6+)

(since this is the current default LTS version for NodeJS)

File: source/includes/quickstart/_index.md [hunk 6]
```diff
@@ -1,142 +1,216 @@
...
-We will show you how to persist a document into kuzzle with the [Javascript SDK](/sdk-documentation)
+It's time to play with the [Kuzzle SDK](/sdk-documentation). In this section, we will persist a document and subscribe to notifications in Kuzzle using the JS SDK.
 
-### Prerequisites
+Before proceeding, ensure that your system matches the following requisites:
 
-- NodeJS, see [instruction here](https://nodejs.org/en/download/) (> v4.3.1)
-- A kuzzle server running
-- That's it
+* A fairly recent version of **NodeJS** (i.e. v4.3.1+) should be installed on your system (<a href="https://nodejs.org/en/download/">instructions here</a>),
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: (i.e. v4.3.1+) => (e.g. v6+)

File: source/includes/quickstart/_index.md [hunk 7]
```diff
@@ -1,142 +1,216 @@
...
 
 ```javascript
 var Kuzzle = require('kuzzle-sdk')
 
-// connection to the kuzzle server
-var kuzzle = new Kuzzle('http://localhost:7512')
+// connect to the kuzzle server
+var kuzzle = new Kuzzle('localhost', {
+    defaultIndex: 'playground',
+    wsPort: 7513
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: `wsPort` is optional and should only be set if the server configuration is different from the defaults.

To keep this tutorial as simple as possible, I would remove this option.

File: source/includes/quickstart/_index.md [hunk 8]
```diff
@@ -1,142 +1,216 @@
...
 
 ```javascript
 var Kuzzle = require('kuzzle-sdk')
 
-// connection to the kuzzle server
-new Kuzzle('http://localhost:7512')
+// connect to the kuzzle server
+var kuzzle = new Kuzzle('localhost', {
+    defaultIndex: 'playground',
+    wsPort: 7513
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: Same as above

File: source/includes/quickstart/_index.md [hunk 9]
```diff
@@ -149,12 +223,32 @@ var filter = {
...
-Try to run the <code>create.js</code> script <a href="#create-your-first-script-to-interact-with-kuzzle">previously created</a> to view the result.
+Congratulations! You just leveraged Kuzzle's pub/sub mechanism.
+</aside>
+
+<aside class="notice">
+Having trouble? <a href="https://gitter.im/kuzzleio/kuzzle-bo">Get in touch with us on Gitter!</a> We'll be happy to help.
 </aside>
+
+### Where do we go from here?
+Now that are fully started and operational with Kuzzle, you can fully leverage its power by:
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: "Now that you are fully [...]"

File: source/includes/quickstart/_index.md (hunk 1 above)
Comment (xbill82) [inline source/includes/quickstart/_index.md]: I definitely agree. I had the same feeling writing it and thought about doing it this way then discarded the idea. :+1:

File: source/includes/quickstart/_index.md (hunk 3 above)
Comment (xbill82) [inline source/includes/quickstart/_index.md]: Yep, much better! And, concerning the response: Kuzzle no logner responds with the Hello message, right? Now it's the routes listing, isn't it?

Comment (xbill82) [inline source/includes/quickstart/_index.md]: oh... just below :sunglasses:

File: source/includes/quickstart/_index.md (hunk 4 above)
Comment (xbill82) [inline source/includes/quickstart/_index.md]: Oh, I liked the ginormous routes list... Do you think it's scary?

Comment (scottinet) [inline source/includes/quickstart/_index.md]: I know I do. Once I even saw a woman in a red dress in it!

File: source/includes/quickstart/_index.md [hunk 10]
```diff
@@ -14,32 +14,28 @@ Before launching a fresh Kuzzle Server, ensure that your system matches the foll
...
 
-Thanks to Docker-compose, running a Kuzzle server is easy. Just grab the standard [docker-compose.yml](https://raw.githubusercontent.com/kuzzleio/kuzzle/master/docker-compose.yml) file and copy it into a directory of your choice.
-
-<aside class="notice">
-Getting a remote file is very easy with curl/wget : <br />
-<code>$ wget https://raw.githubusercontent.com/kuzzleio/kuzzle/master/docker-compose.yml</code>
-</aside>
+Thanks to Docker-compose, running a Kuzzle server is easy. Just grab the standard [docker-compose.yml](https://raw.githubusercontent.com/kuzzleio/kuzzle/master/docker-compose.yml) file, copy it into a directory and start Kuzzle:
 
 Then, run
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: Leftover :-)

File: source/includes/quickstart/_index.md [hunk 11]
```diff
@@ -14,32 +14,28 @@ Before launching a fresh Kuzzle Server, ensure that your system matches the foll
...
 
-To check that the Kuzzle Server is actually running, you can hit the main HTTP API endpoint by browsing the page <a href="http://localhost:7511/api/1.0">http://localhost:7511/api/1.0</a> or via cURL on the command line:
+Your Kuzzle instance is now ready to be used. For instance, you can hit the main HTTP API endpoint by browsing the page <a href="http://localhost:7511">http://localhost:7511</a> or via cURL on the command line:
 
 ```bash
 $ curl "http://localhost:7511/"
 ```
 
-Kuzzle will respond you with a friendly hello message :)
+Kuzzle will respond you with a list of the existing routes.
```
Comment (scottinet) [inline source/includes/quickstart/_index.md]: Wow, you're really serious about keeping it!

Well... At least, add a "?pretty=true" to the URL, unless you like walls of text.

**Comment by xbill82** [comment]
Closing due to splitting into #86 and #85
